#!/usr/bin/env python
PACKAGE = "cacc_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("vel_ref", double_t, 0, "Reference speed in mps", 1.0, 0.0, 20.0)
gen.add("ds", double_t, 0, "Ds", 0.5, 0.0, 20.0)
gen.add("dmin", double_t, 0, "Dmin", 0.3, 0.0, 20.0)
gen.add("umax", double_t, 0, "Umax", 1.0, 0.0, 3.0)
gen.add("umin", double_t, 0, "Umin", 3.0, 0.0, 6.0)
gen.add("q0", int_t, 0, "Q0", 30, 0, 100)
gen.add("q1", int_t, 0, "Q1", 1, 0, 100)
gen.add("q2", int_t, 0, "Q2", 1, 0, 100)
gen.add("r", int_t, 0, "R", 800, 0, 2000)
gen.add("sf", int_t, 0, "Sf", 100, 0, 1000)
gen.add("su", int_t, 0, "Su", 1000, 0, 1000)
gen.add("sd", int_t, 0, "Sd", 1000, 0, 1000)

exit(gen.generate(PACKAGE, "mpc_parameter_configuration", "MpcParam"))
